File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L1/08/Level1.ts
import { Scanner } from 'readline-sync';

/**
 *
 * @author FD8DBE9073AFCC0504CD6901E1091CAD
 */
function main(): void {
    const input: Scanner = new Scanner();
    console.log('Enter a 4 by 4 matrix row by row: ');
    const m: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));
    for (let i = 0; i < 4; i++) {
        for (let j = 0; j < 4; j++) {
            m[i][j] = input.nextDouble();
        }
    }
    console.log(`Sum of the elements in the major diagonal is ${sumMajorDiagonal(m)}`);
}

/**
 * Function to sum the major diagonal
 * @param m the 4x4 matrix
 * @returns the sum of the major diagonal
 */
function sumMajorDiagonal(m: number[][]): number {
    let sum = 0;
    for (let i = 0; i < m.length; i++) {
        sum += m[i][i];
    }
    return sum;
}

main();

File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L1/08/Level1.kt
package IR

import java.util.*

/*
* To change this license header, choose License Headers in Project Properties.
* To change this template file, choose Tools | Templates
* and open the template in the editor.
*/
/**
 *
 * @author FD8DBE9073AFCC0504CD6901E1091CAD
 */
object Level1 {
    @JvmStatic
    fun main(args: Array<String>) {
        val input = Scanner(System.`in`) //harus import dulu
        print("Enter a 4 by 4 matrix row by row: ")
        val m = Array(4) { DoubleArray(4) }
        for (i in 0..3) {
            for (j in 0..3) {
                m[i][j] = input.nextDouble()
            }
        }
        print(
            "Sum of the elements in the major diagonal is "
                    + sumMajorDiagonal(m)
        )
    }

    //fungsi untuk menjumlahkan diagonal
    fun sumMajorDiagonal(m: Array<DoubleArray>): Double {
        var sum = 0.0
        for (i in m.indices) {
            sum += m[i][i]
        }
        return sum
    }
}
